### PR TITLE
Closes #1534 Adding multi-byte character support to EntityEmbedProcess migrate plugin.

### LINKED ISSUE #1534: Text with certain special characters is improperly migrated
## Problem/Motivation
UAQS pages may have text that contains special characters, some of which gets imported improperly.

### Describe the bug
The particular case I found is a UAQS Flexible Page with a Text area paragraph that contains a registered mark ® (e.g., Spacewatch®) results in it being migrated to an AZQS Page with a Text paragraph that replaces that registered mark with Â® (e.g., SpacewatchÂ®),

In comparison, basic pages or other custom content types that were created in Drupal 7 and migrated using Drupal 9's built-in migration tools get imported with correct text.

### PR DESCRIPTION
**Pull request by trackleft**
## Description

Using mb_convert_encoding to convert field value to UTF-8 before passing the field value to \DOMDocument().

References:
- https://www.php.net/manual/en/function.mb-convert-encoding.php
- https://github.com/Masterminds/html5-php/blob/master/src/HTML5/Parser/UTF8Utils.php 
- https://git.drupalcode.org/project/migrate_plus/-/blob/6.0.x/src/Plugin/migrate/process/Dom.php

## Related issues
Closes #1534 

## How to test

Source site migration must have text areas with multibyte strings.

### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

### Drupal core
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

### Drupal contrib projects
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by joeparsons — APPROVED** [review]
Looks good to me.

**Comment by trackleft** [comment]
Possibly add 
```
iconv('UTF-8', 'ASCII//TRANSLIT', $input);
```

File: modules/custom/az_migration/src/Plugin/migrate/process/EntityEmbedProcess.php [hunk 1]
```diff
@@ -208,6 +208,13 @@ public function updateEmbedTag($id, $type, $tag, $view, \DOMDocument $dom, \DOME
    */
   public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
 
+    // Return $value if there are no <drupal-entity> elements.
+    if (strpos($value, '<drupal-entity ') === FALSE) {
+      return $value;
+    }
+    // Convert $value to UTF-8.
+    $value = mb_convert_encoding($value, 'UTF-8');
```
Comment (tadean) [inline modules/custom/az_migration/src/Plugin/migrate/process/EntityEmbedProcess.php:216]:
```suggestion
    $value = mb_convert_encoding($value, 'HTML-ENTITIES', 'UTF-8');
```

File: modules/custom/az_migration/src/Plugin/migrate/process/EntityEmbedProcess.php [hunk 2]
```diff
@@ -208,6 +208,13 @@ public function updateEmbedTag($id, $type, $tag, $view, \DOMDocument $dom, \DOME
...
+
     $dom = new \DOMDocument();
```
Comment (tadean) [inline modules/custom/az_migration/src/Plugin/migrate/process/EntityEmbedProcess.php:218]:
```suggestion
    $dom = new \DOMDocument('1.0', 'UTF-8');
```

**Review by joeparsons — APPROVED** [review]

**Comment by joeparsons** [comment]
I think merging main into this branch will fix the PHPstan errors.

**Comment by trackleft** [comment]
I don't think the migration error is related  `[notice] Processed 49 items (44 created, 0 updated, 5 failed, 0 ignored) - done with 'az_files'`

**Review by joeparsons — APPROVED** [review]
Looks good to me.  All the checks passed now.

**Review by tadean — APPROVED** [review]

**Pull request merged**